grafana-plugin/src/containers/RemoteSelect/RemoteSelect.tsx

```diff
@@ -58,14 +56,24 @@ const RemoteSelect = inject('store')(
       }));
     };
 
+    function mergeOptions(oldOptions: SelectableValue[], newOptions: SelectableValue[]) {
+      const existedValues = oldOptions.map((o) => o.value);
+      return newOptions.filter(({ value }) => !existedValues.includes(value)).concat(oldOptions);
+    }
+
+    const [options, setOptions] = useReducer(mergeOptions, []);
+
     useEffect(() => {
       makeRequest(href, {}).then((data) => {
         setOptions(getOptions(data.results || data));
       });
     }, []);
 
     const loadOptionsCallback = useCallback((query: string) => {
-      return makeRequest(href, { params: { search: query } }).then((data) => getOptions(data.results || data));
+      return makeRequest(href, { params: { search: query } }).then((data) => {
+        setOptions(getOptions(data.results || data));
```

Comment:
this `loadOptionsCallback` should return old options +  new options (merged together), but it works in some other weird way

Reply:
Had a call with @maskin25 and decided to leave it as it is